### ISSUE TITLE
False positive on object destructuring in arrow function parameters

### ISSUE THREAD

**Issue by davidwkeith** [getify/eslint-plugin-proper-arrows#24]
The following will report a params count error:

```javascript
({foo, bar, baz, bat, bongo }) => {/* a function that handles all those 'named' parameters */}
```

While the following won't
```
function namedParams({foo, bar, baz, bat, bongo }) {/* a function that handles all those 'named' parameters */}
```

This might be intentional, as it seems weird to use the pattern in an anonymous function, but the lint error should be more exact.

**Comment by getify** [comment]
This plugin is called "proper-arrows" and thus only applies to arrow functions, not all general functions. There are many built-in ESLint rules which apply rules to functions generally, but I wanted this plugin to apply rules specifically for arrow functions.

In this specific case, this "max-params" built-in ESLint rule can be used: https://eslint.org/docs/rules/max-params#enforce-a-maximum-number-of-parameters-in-function-definitions-max-params

This plugin's params-count rule specifically is intended to be used when you want to apply a limit **ONLY** to arrow functions, with a different (or no) limit applicable for non-arrow functions. If you just want the same params-count applied for all functions, disable this plugin rule and use the built-in "max-params" rule instead.

**Comment by davidwkeith** [comment]
Makes sense, but technically there is only one argument, maybe breaking both of these rules into two parts, arguments and pseudo named arguments. Named arguments are easier to reason about and should be encouraged for when a longer method signature is required. (though really shouldn't be an arrow function then)

I will play around with the main ESLint rule and see if I can configure it to do what I want. (fewer unnamed parameters, and a reasonable, but larger amount of "named" parameters.)

**Comment by getify** [comment]
Closing for now.
